contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -11,6 +11,13 @@ RETURNS ntext
 AS 'babelfishpg_tsql', 'tsql_query_to_xml_text'
 LANGUAGE C IMMUTABLE STRICT COST 100;
 
+-- Helper function to support the FOR JSON clause
+CREATE OR REPLACE FUNCTION sys.tsql_query_to_json_text(query text, mode int, include_null_value boolean,
+           without_array_wrappers boolean,root_name_present boolean, root_name text)
```

Comment:
Yes, it can be possible that the root_name is passed as empty string. As in the TsqlForJSONMakeFuncCall, when converting the input select query into function call tsql_query_to_json_text(), I am passing NULL root_name as empty string, therefore I need a bool variable to check if root_name is present.